main.tf

```diff
@@ -238,7 +238,7 @@ resource "aws_rds_cluster" "secondary" {
 }
 
 resource "aws_rds_cluster_instance" "default" {
-  count                                 = local.cluster_instance_count
```

Comment:
@matharoo Sorry, but help me understand this a bit more: How will any aurora usage of this module create their corresponding instance if we're explicitly setting this to 0? Is this a long term fix or will this become unnecessary at some point?

Reply:
Hi @Gowiem , this is not setting it to 0. In `main.tf` at the beginning of file where we decalre locals, we can see its taking the default value of `2` coming from `var.cluster_size`. 

```
locals {
  enabled = module.this.enabled
.....
  cluster_instance_count   = local.enabled ? var.cluster_size : 0  <--
 .......
}
```

With my change It only sets it to 0 when the engine is not aurora as checked by the regex that I put in place. Based on the documentation of [aws_rds_cluster](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/rds_cluster) :

> To manage cluster instances that inherit configuration from the cluster (when not running the cluster in serverless engine mode), see the [aws_rds_cluster_instance resource](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/rds_cluster_instance).

So this is a necessary change that will only allow the `aws_rds_cluster_instance` to be created if engine is aurora for non-aurora it will not create this resource and not error out as explained in the PR description.

Reply:
It’s very possible that I’m being thick here, so bear with me, but my question still stands: how will a non-aurora RDS cluster have any instances if we’re setting `aws_rds_cluster_instance`’s `count` to `0`? Are you trying to get at that non-aurora cluster’s don’t need instances because they’re single instance and creating only the cluster will somehow suffice? What am I missing here?

Reply:
Pulling in @nitrocode as a second set of eyes.

Reply:
ah sorry i misinterpret your comment, In my test example i was trying to setup a multi a-z postgres rds cluster and i found that we cannot add instances to these type of clusters. By default when we setup a multi a-z cluster using `aws_rds_cluster` resource it sets up one writer instance and 2 readers all in 3 different a-z's. At present AWS does not support adding more instances to this type of multi a-z cluster. I couldn't find it on AWS console as well as on the AWS terraform provider code. The only place i found this mentioned was when trying to setup cluster manually in AWS console:

![image](https://user-images.githubusercontent.com/5571933/180856133-646f3472-12ef-479e-8ef9-706c5f99ca99.png)


 The resource `aws_rds_cluster_instance` is only to add instances to aurora-clusters.

Reply:
@matharoo Hm... after looking at the docs, I'm wondering if you're using this module incorrectly. You might be better suited to just using https://github.com/cloudposse/terraform-aws-rds which seems more suited to your needs. 

The below is from the AWS Provider documentation for the [`aws_rds_cluster`](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/rds_cluster) resource: 

> Manages a [RDS Aurora Cluster](https://docs.aws.amazon.com/AmazonRDS/latest/UserGuide/CHAP_Aurora.html). To manage cluster instances that inherit configuration from the cluster (when not running the cluster in serverless engine mode), see the [aws_rds_cluster_instance resource](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/rds_cluster_instance). To manage non-Aurora databases (e.g., MySQL, PostgreSQL, SQL Server, etc.), see the [aws_db_instance resource](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/db_instance).

Because it specifies that you should use `aws_db_instance` for non-Aurora databases... I feel like that is the correct route to take. 

Let me know your thoughts and if there is something I'm missing here.

Reply:
Yes but what `aws_db_instance` will be launching a standalone db and cannot be used in conjunction with `aws_rds_cluster` resource which is what i found after going over the documentation unless i missed something. Our requirement is to setup a cluster that comes with perks of replication and reliability of azs whereas the setup of standalone dbs to create a cluster gets complex. I thought this module was for all kinds of AWS RDS clusters but i am assuming the aim is to go for aurora only if i am not mistaken.

This module is working fine except for non-aurora multi-az clusters that only allow 3 nodes(one writer and 2 readers), where as in case of aurora `aws_rds_cluster_instance` resource allows adding more. 

To be inline with what AWS currently supports for non-aurora multi a-z clusters, my change seems to work fine without the error as shown description of this PR and i was successfully able to spin up that cluster using this module. And it doesn't seem to break any of the functionality of the module.

<img width="1384" alt="image" src="https://user-images.githubusercontent.com/5571933/181386465-088a6a5f-ce57-43fd-939b-30a3716b4597.png">

Reply:
@matharoo I gotcha. Let me bring this up with the maintainer team and see what folks sentiments are surrounding this issue. I'll report back in a day or two.